app/script/main/app.js

```diff
@@ -559,48 +559,18 @@ z.main.App = class App {
    * Check that this is the single instance tab of the app.
    * @returns {Promise} Resolves when page is the first tab
    */
-  _checkSingleInstanceOnInit() {
-    if (!z.util.Environment.electron) {
-      return this._setSingleInstanceCookie();
-    }
-
-    return Promise.resolve();
-  }
-
-  _checkSingleInstanceOnInterval() {
-    const singleInstanceCookie = Cookies.getJSON(App.CONFIG.TABS_CHECK.COOKIE_NAME);
-
-    const shouldBlockTab = !singleInstanceCookie || singleInstanceCookie.appInstanceId !== this.instanceId;
-    if (shouldBlockTab) {
-      return this._redirectToLogin(z.auth.SIGN_OUT_REASON.MULTIPLE_TABS);
+  _registerSingleInstance() {
+    if (this.singleInstanceHandler.registerInstance(this.instanceId)) {
+      this._registerSingleInstanceCleaning();
+      return Promise.resolve();
     }
+    return Promise.reject(new Error('App already running in another instance'));
```

Comment:
Maybe we should even move the Promise based signature to the handler

Reply:
Yep, good idea

Reply:
Humm actually, I have second thoughts about this idea.
It seems overkill. If the argument for this is that 'it is convenient for the `App` class, then I think it's not a valid one. 
I think we should use the most primitive types possible for those leaf handlers. 
I would say if it is not async, then no need for promises.
WDYT?